pages/shared.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from "react";
+import Banner from "../components/domains/shared/Banner";
+import CardList from "../components/commons/CardList";
+import SearchInput from "../components/commons/SearchInput";
+import styles from "../styles/sharedPage.module.css";
+import { getSharedData } from "./api/SharedApi";
+import { Folder } from "../types/shared";
+import { Link } from "../types/common";
+
+function SharedPage({ openModal }: { openModal: () => void }) {
+  const [searchResult, setSearchResult] = useState<string>("");
+  const [sharedFolder, setSharedFolder] = useState<Folder>({
+    id: 1,
+    name: "",
+    owner: {
+      id: 2,
+      name: "",
+      profileImageSource: "",
+    },
+    links: [
+      {
+        id: 1,
+        url: "",
+        title: "",
+        description: "",
+        image_source: "",
+        created_at: "",
+      },
+    ],
+  });
+
+  const handleFolderLoad = async () => {
+    const { folder } = await getSharedData();
+    setSharedFolder({
+      ...folder,
+      links: folder.links,
+    });
+  };
+
+  const searchLink = async (keyword: string) => {
+    const { folder } = await getSharedData();
+    const filteredLinks = folder.links.filter(
+      (link: Link) =>
+        link.url?.includes(keyword) ||
+        link.title?.includes(keyword) ||
+        link.description?.includes(keyword)
+    );
+
+    setSharedFolder({
+      ...folder,
+      links: filteredLinks,
+    });
+  };
+
+  useEffect(() => {
+    handleFolderLoad();
+  }, []);
+
+  return (
+    <>
+      <Banner folder={sharedFolder} />
+      <section className={styles.contentFlax}>
+        <div className={styles.contentBox}>
+          <SearchInput
+            searchResult={searchResult}
+            setSearchResult={setSearchResult}
+            searchLink={searchLink}
```

Comment:
SearchInput에게 검색기능을 가진 콜백을 전달하는 것은 자연스러워 보입니다. 👍  
다만 prop이름도 이벤트핸들러 답게 네이밍하면 좋겠습니다
```suggestion
            onSearch={searchLink}
```